assets/src/stores/amp-story/reducer.js

```diff
@@ -50,14 +53,42 @@ export function animationOrder( state = {}, action ) {
 				[ page ]: pageAnimationOrder,
 			};
 
-		case 'REMOVE_ANIMATION':
-			if ( entryIndex( item ) !== -1 ) {
-				const itemPredecessor = pageAnimationOrder[ entryIndex( item ) ].parent;
-				pageAnimationOrder[ entryIndex( item ) ].parent = undefined;
+		case 'CHANGE_ANIMATION_TYPE':
+			// Animation was disabled, update all successors.
+			if ( ! animationType ) {
+				if ( entryIndex( item ) !== -1 ) {
+					const itemPredecessor = pageAnimationOrder[ entryIndex( item ) ].parent;
+
+					for ( const successor in pageAnimationOrder.filter( ( { parent: p } ) => p === item ) ) {
+						pageAnimationOrder[ successor ].parent = itemPredecessor.parent;
+					}
 
-				for ( const successor in pageAnimationOrder.filter( ( { parent: p } ) => p === item ) ) {
-					pageAnimationOrder[ successor ].parent = itemPredecessor.parent;
+					pageAnimationOrder.splice( entryIndex( item ), 1 );
 				}
+			} else if ( entryIndex( item ) !== -1 ) {
+				pageAnimationOrder[ entryIndex( item ) ].animationType = animationType;
+			} else {
+				pageAnimationOrder.push( { id: item, animationType } );
+			}
+
+			return {
+				...newAnimationOrder,
+				[ page ]: pageAnimationOrder,
+			};
+
+		case 'CHANGE_ANIMATION_DURATION':
```

Comment:
If the duration or delay changes, would the Advance time of the relevant Page also change in case of `time` setting and in case the new values exceed the chosen value? E.g. if the user has previously chosen to advance in 1 sec but changes the animation settings so that the total animation time reaches 3 sec, would the lower value 1 sec automatically change?

Reply:
Not currently, but I guess it makes sense

Reply:
Done in f25b433ad43b094f732c122d96d943e35d7fee9a